pkg/bmc/access_test.go

```diff
@@ -176,338 +196,258 @@ func TestParse(t *testing.T) {
 				t.Fatalf("unexpected parse error: %v", err)
 			}
 
-			if T != tc.Type {
-				t.Fatalf("expected type %q but got %q", tc.Type, T)
+			if url.Scheme != tc.Type {
+				t.Fatalf("expected type %q but got %q", tc.Type, url.Scheme)
 			}
 
-			if P != tc.Port {
-				t.Fatalf("expected port %q but got %q", tc.Port, P)
+			if url.Port() != tc.Port {
+				t.Fatalf("expected port %q but got %q", tc.Port, url.Port())
 			}
 
-			if H != tc.Host {
-				t.Fatalf("expected host %q but got %q", tc.Host, H)
+			if url.Hostname() != tc.Host {
```

Comment:
It's unfortunate that the test data maps url.Hostname() to tc.Host and url.Host to tc.Hostname. We can live with it for now, and maybe fix it in another PR.

Reply:
LOL.
Personally I don't think it even necessary to check url.Host. If url.Hostname() and url.Port() are right then it can only be because we got the right data.

Reply:
ok, I can create another PR later to fix this. If this is ok and gets merged then we can focus on #284 . I'll do the fix after. Thank you